middleware.ts

```diff
@@ -0,0 +1,8 @@
+import authConfig from "./auth.config"
```

Comment:
Add a test that attempts to make a request to the backend API to test that routes are secure under this middleware.

Reply:
Would this work with the test?

```
/**
 * @jest-environment node
 */
import '@testing-library/jest-dom'
import { middleware } from '../middleware';

describe('Middleware', () => {
    afterEach(() => {
        middleware.mockReset();
      });
    it('Testing if endpoint is secured', async () => {
        const User = {
            id: '1',
            name: 'John Doe',
            email: 'example@example.com',
            image: 'example'
        }
        await middleware(User)
        expect(middleware).toHaveBeenCalledTimes(1)
        expect(middleware).toHaveBeenCalledWith(User)
    })

    it('Testing if endpoint is not secured', async () => {
        const User = {}
        await middleware(User)
        expect(middleware).toHaveBeenCalledTimes(1)
        expect(middleware).toHaveBeenCalledWith(User)
    })
})
```

Reply:
Let's skip testing for the middleware for now. It's more beneficial for you to have a project that you can present rather than focusing on this anyway. Fyi, both of the tests you wrote here solely test if (1) the middleware function takes an object and (2) if it was called. However, both of these scenarios happen within the test setup and don't measure the functionality of the middleware. This means that I can substitute middleware for another function `exampleFunction` that takes an object and still make the tests pass. The goals are to (1) test that the middleware receives requests from the frontend, (2) test that auth is applied to these requests.

Reply:
yeah you're definitely right, I couldn't find any information how this runs

src/middleware.ts
```
export const { auth: middleware } = NextAuth(authConfig)
```
in the backend on authjs. Unless, I used my own custom middleware function which redirects to a custom login page.